connect/runtime/src/test/java/org/apache/kafka/connect/runtime/rest/RestServerTest.java

```diff
@@ -79,6 +83,72 @@ public void testCORSDisabled() {
         checkCORSRequest("", "http://bar.com", null, null);
     }
 
+    @Test
+    public void testParseListeners() {
+        // Use listeners field
+        Map<String, String> configMap = new HashMap<>(baseWorkerProps());
+        configMap.put(WorkerConfig.LISTENERS_CONFIG, "http://localhost:8080,https://localhost:8443");
+        DistributedConfig config = new DistributedConfig(configMap);
+
+        server = new RestServer(config);
+        Assert.assertArrayEquals(new String[] {"http://localhost:8080", "https://localhost:8443"}, server.parseListeners().toArray());
+
+        // Build listener from hostname and port
+        configMap = new HashMap<>(baseWorkerProps());
+        configMap.put(WorkerConfig.REST_HOST_NAME_CONFIG, "my-hostname");
+        configMap.put(WorkerConfig.REST_PORT_CONFIG, "8080");
+        config = new DistributedConfig(configMap);
+        server = new RestServer(config);
+        Assert.assertArrayEquals(new String[] {"HTTP://my-hostname:8080"}, server.parseListeners().toArray());
```

Comment:
hmm, seems odd that we're getting a mix of parsed values that has both `HTTP` (all caps) and `http` (lowercase, from above). should these end up all standardized to one or the other?